pkg/apis/application/v1alpha1/applicationset_types.go

```diff
@@ -589,6 +589,8 @@ type SCMProviderGeneratorFilter struct {
 	LabelMatch *string `json:"labelMatch,omitempty" protobuf:"bytes,4,opt,name=labelMatch"`
 	// A regex which must match the branch name.
 	BranchMatch *string `json:"branchMatch,omitempty" protobuf:"bytes,5,opt,name=branchMatch"`
+	// Include archived repos.
+	IncludeArchivedRepos bool `json:"includeArchivedRepos,omitempty" protobuf:"varint,6,opt,name=includeArchivedRepos"`
```

Comment:
I agree with @agaudreault's [comment](https://github.com/argoproj/argo-cd/issues/18531#issuecomment-2183576125). When using booleans and defaults, it is easier to have the default be false.